test/OpenTelemetry.Extensions.Docker.Tests/Resources/DockerResourceDetectorTests.cs

```diff
@@ -57,50 +57,149 @@ public class DockerResourceDetectorTests
     private const string CONTAINERID =
         "d86d75589bf6cc254f3e2cc29debdf85dde404998aa128997a819ff991827356";
 
+    // cgroupv2 line with container Id
+    private const string CGROUPLINEV2 =
+        "13:name=systemd:/pod/d86d75589bf6cc254f3e2cc29debdf85dde404998aa128997a819ff991827356/hostname";
+
+    // cgroupv2 Expected Container Id 
+    private const string CONTAINERIDV2 = "d86d75589bf6cc254f3e2cc29debdf85dde404998aa128997a819ff991827356";
+
+    // hostname
+    private const string CGROUPLINEV2WITHHOSTNAME =
+        "473 456 254:1 /docker/containers/dc64b5743252dbaef6e30521c34d6bbd1620c8ce65bdb7bf9e7143b61bb5b183/hostname /etc/hostname rw,relatime - ext4 /dev/vda1 rw";
+
+    private const string CONTAINERIDV2WITHHOSTNAME = "dc64b5743252dbaef6e30521c34d6bbd1620c8ce65bdb7bf9e7143b61bb5b183";
+
+    // minikube containerd mountinfo
+    private const string CGROUPLINEV2WITHMINIKUBECONTAINERD =
+        "1537 1517 8:1 /var/lib/containerd/io.containerd.grpc.v1.cri/sandboxes/fb5916a02feca96bdeecd8e062df9e5e51d6617c8214b5e1f3ff9320f4402ae6/hostname /etc/hostname rw,relatime - ext4 /dev/sda1 rw";
+
+    private const string CONTAINERIDV2WITHMINIKUBECONTAINERD = "fb5916a02feca96bdeecd8e062df9e5e51d6617c8214b5e1f3ff9320f4402ae6";
+
+    // minikube docker mountinfo
+    private const string CGROUPLINEV2WITHDOCKER =
+        "2327 2307 8:1 /var/lib/docker/containers/a1551a1d7e1881d6c18d2c9ec462cab6ad3666825f0adb2098e9d5b198fd7e19/hostname /etc/hostname rw,relatime - ext4 /dev/sda1 rw";
+
+    private const string CONTAINERIDV2WITHDOCKER = "a1551a1d7e1881d6c18d2c9ec462cab6ad3666825f0adb2098e9d5b198fd7e19";
+
+    // minikube docker mountinfo2
+    private const string CGROUPLINEV2WITHDOCKER2 =
+        "929 920 254:1 /docker/volumes/minikube/_data/lib/docker/containers/0eaa6718003210b6520f7e82d14b4c8d4743057a958a503626240f8d1900bc33/hostname /etc/hostname rw,relatime - ext4 /dev/vda1 rw";
+
+    private const string CONTAINERIDV2WITHDOCKER2 = "0eaa6718003210b6520f7e82d14b4c8d4743057a958a503626240f8d1900bc33";
+
+    // podman mountinfo
+    private const string CGROUPLINEV2WITHPODMAN =
+        "1096 1088 0:104 /containers/overlay-containers/1a2de27e7157106568f7e081e42a8c14858c02bd9df30d6e352b298178b46809/userdata/hostname /etc/hostname rw,nosuid,nodev,relatime - tmpfs tmpfs rw,size=813800k,nr_inodes=203450,mode=700,uid=1000,gid=1000";
+
+    private const string CONTAINERIDV2WITHPODMAN = "1a2de27e7157106568f7e081e42a8c14858c02bd9df30d6e352b298178b46809";
+
+    // Invalid cgroup line (contains a z)"
+    private const string INVALIDHEXCGROUPLINEV2 =
```

Comment:
We have so many const defined in that file now. I like @pellared suggestion to look into structure of tests in https://github.com/open-telemetry/opentelemetry-go/pull/3508 - they looks nice with a dictionary of named cased with input / expected result pairs.

Reply:
I can also turn it into a theory with InlineData if that works better in this context